### PR TITLE
bootstrap default keyspace group in the tso service

### PR DESCRIPTION
**Pull request by binshi-bing**
### What problem does this PR solve?

Issue Number: Ref #6232

### What is changed and how does it work?

```commit-message
Changes:

1. Introduce the initialization logic of the default keyspace group.
- If the default keyspace group isn't configured in the etcd, every tso node/pod should initialize it and join the election for the primary of this group.
- If the default keyspace group is configured in the etcd, the tso nodes/pods which are assigned with this group will initialize it and join the election for the primary of this group.

2. Introduce the keyspace group membership restriction -- default keyspace always belongs to default keyspace group.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/keyspace/tso_keyspace_group.go [hunk 1]
```diff
@@ -71,15 +76,8 @@ func (m *GroupManager) Bootstrap() error {
 		return err
 	}
 
-	userKind := endpoint.StringUserKind(defaultKeyspaceGroup.UserKind)
```
Comment (binshi-bing) [inline pkg/keyspace/tso_keyspace_group.go:74]: delete line 74 ~ 79 for two reasons:
1. This is covered by the for loop below in which it loads all keyspace groups, including the default keyspace group, and initialize m.groups.
2. Theoretically,  at line 79, we put defaultKeyspaceGroup into m.groups, but defaultKeyspaceGroup might be different from the latest version persisted in the store, though later on the current logic of indexedHeap will overwrite the existing one if put a new one.

Comment (binshi-bing) [reply pkg/keyspace/tso_keyspace_group.go:74]: @rleungx ^

Comment (binshi-bing) [reply pkg/keyspace/tso_keyspace_group.go:74]: Let me know if I missed anything.

Comment (rleungx) [reply pkg/keyspace/tso_keyspace_group.go:74]: You're right.

File: pkg/tso/keyspace_group_manager.go [hunk 2]
```diff
@@ -313,10 +320,15 @@ func (kgm *KeyspaceGroupManager) initAssignment(ctx context.Context) (int64, err
 		for _, group := range groups {
 			select {
 			case <-ctx.Done():
-				return watchStartRevision, errs.ErrLoadKeyspaceGroupsTerminated
+				err = errs.ErrLoadKeyspaceGroupsTerminated
```
Comment (rleungx) [inline pkg/tso/keyspace_group_manager.go:323]: I think we can return nil here instead of an error.

**Review by JmPotato — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**